machine.go

```diff
@@ -1067,3 +1071,57 @@ func (m *Machine) CreateSnapshot(ctx context.Context, memFilePath, snapshotPath
 	m.logger.Debug("snapshot created successfully")
 	return nil
 }
+
+// AttachBalloon creates a balloon device if one does not exist
+func (m *Machine) AttachBalloon(ctx context.Context, dev models.Balloon) error {
+	// TODO: Creates a new balloon device if one does not already exist, otherwise updates it, before machine startup.
+    // This will fail after machine startup.
+	if dev.AmountMb == nil {
```

Comment:
How about returning an error when AmountMb is nil?

Reply:
Fixed it.

Reply:
No. Let's make it work for the released version first.

Reply:
Just to make sure we gonna use AmountMib, right?